dev/com.ibm.ws.kernel.feature.core/src/com/ibm/ws/kernel/feature/internal/FeatureManager.java

```diff
@@ -1891,10 +1895,25 @@ boolean reportErrors(Result result, Collection<String> restrictedAccessAttempts,
             resolved.clear();
             Tr.warning(tc, "UPDATE_DISABLED_FEATURES_ON_CONFLICT");
         }
+
         return reportedErrors;
 
     }
 
+    /**
+     * If server launch was requested with a checkpoint, and checkpoint feature is not enabled an
+     * error message is issued
+     */
+    private void CheckiFCheckpointFeatureMissing(Collection<String> features) {
+
+        //TODO remove beta check on feature release.
+        if (CheckpointPhase != null && ProductInfo.getBetaEdition()) {
+            if (!features.contains("checkpoint-1.0")) {
+                Tr.error(tc, "CHECKPOINT_REQUESTED_CHECKPOINT_FEATURE_MISSING", CheckpointPhase);
```

Comment:
Should we shutdown the server here?  I would say yes, but only if the output to the console gives adequate information to the user that something went wrong.  Otherwise it may look like everything went fine until the user discovers no image.  I supposed the error message itself will have been displayed in the console.  We need to ensure the server script returns non 0 return code when things go wrong so that scripts that do a checkpoint action can also fail appropriately when a checkpoint does not succeed.